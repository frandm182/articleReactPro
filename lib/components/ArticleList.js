import React from 'react';
import Article from './Article';

const ArticleList = (props) => (
  <div>
    {Object.values(props.articles).map((article) => 
      <Article
        key={article.id}
        article={article} 
        author={props.authors[article.authorId]}/>
    )}
  </div>);
export default ArticleList;