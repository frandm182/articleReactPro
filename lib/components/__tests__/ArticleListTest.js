import React from 'react';
import 'babel-polyfill';
import ArticleList from '../ArticleList';
import ArticleContainer from '../Article';

// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' }
    }
  };
  ArticleContainer.propTypes = {};
  it('renders correctly', () => {
    const wrapper = shallow(<ArticleList {...testProps} />);
    expect(wrapper.find('ArticleContainer').length).toBe(2);

    expect(wrapper).toMatchSnapshot();
  });
});