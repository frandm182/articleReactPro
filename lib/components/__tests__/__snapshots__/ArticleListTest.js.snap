// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <ArticleContainer
    article={
      Object {
        "id": "a",
      }
    }
    key="a"
  />
  <ArticleContainer
    article={
      Object {
        "id": "b",
      }
    }
    key="b"
  />
</div>
`;
